### DISCUSSION TITLE
How to solve issue calling spacy-cpp multiple times sequentially?

### DISCUSSION THREAD

**Discussion by VirArman** [d99kris/spacy-cpp#24] [Q&A]
When I try to create spacy object in for loop during the second iteration the spacy constructor does not see the spacy import path.
Tried to reproduce same error in spacy python version and the code worked without any problem.
I would like to know the possible cause or solution for this problem 

int main()
{
    for(int i = 0; i < 2; ++i ) { 
        Spacy::Spacy spacy;
        auto nlp = spacy.load("en_core_web_sm");
        auto doc0 = nlp.parse("he fire his house");
            for (auto& token : doc0.tokens())
            {
                  std::cout << token.text() << " [" << token.pos_() << "],";
                  std::cout << "[" << token.lemma_() << "]," << "\n";
            }
            std::cout << std::endl;
    }
}

**Comment by d99kris** [comment]
Hi @VirArman - spacy-cpp is implemented as a simple wrapper around the python library spacy, so all calls goes through a Python run-time and some interfacing like library import are not exactly the same as when using a Python interpreter. Ideally your code example should work, but I think I'll need to exclude it from the scope of spacy-cpp. You may work around the issue by using a single Spacy instance, for example:
```c++
int main()
{
    Spacy::Spacy spacy;
    for(int i = 0; i < 2; ++i ) {
        auto nlp = spacy.load("en_core_web_sm");
        auto doc0 = nlp.parse("he fire his house");
            for (auto& token : doc0.tokens())
            {
                  std::cout << token.text() << " [" << token.pos_() << "],";
                  std::cout << "[" << token.lemma_() << "]," << "\n";
            }
            std::cout << std::endl;
    }
}
```
You'll likely also want to bring out the `spacy.load()` call outside the loop (assuming you're just using a single language model), as it's a heavy operation.

**Comment by VirArman** [comment]
Thanks @d99kris, the solution you offered worked
